sklearn/datasets/_openml.py

```diff
@@ -457,15 +457,16 @@ def _download_data_arff(file_id, sparse, data_home, encode_nominal=True):
 
     @_retry_with_clean_cache(url, data_home)
     def _arff_load():
-        with closing(_open_openml_url(url, data_home)) as response:
-            if sparse is True:
-                return_type = _arff.COO
-            else:
-                return_type = _arff.DENSE_GEN
-
-            arff_file = _arff.loads(response.read().decode('utf-8'),
-                                    encode_nominal=encode_nominal,
-                                    return_type=return_type)
+        response = _open_openml_url(url, data_home)
+        if sparse is True:
+            return_type = _arff.COO
+        else:
+            return_type = _arff.DENSE_GEN
+
+        arff_file = _arff.load((line.decode('utf-8')
+                                for line in response),
```

Comment:
@shashanksingh28 if this PR is merged before #14800 I'd suggest just making a `_check_md5` helper which takes an iterable of bytes as input and returns generates iterable of bytes, checking md5 on the way.

Reply:
Sounds good. I would rather wait for this to go in first and merge the md5 after.